src/components/Home/Home.jsx

```diff
@@ -3,15 +3,36 @@ import UserCard from "./UserCard/UserCard";
 import { useNavigate } from "react-router-dom";
 import { useState } from "react";
 import "./Home.css";
+import Filter from "./Filter/Filter";
 
 const Home = ({ users }) => {
+
   const [searchValue, setSearchValue] = useState("");
+  const [skillsFilterValue, setSkillsFilterValue] = useState("");
+  const [typesFilterValue, setTypesFilterValue] = useState("");
+
   const navigate = useNavigate();
 
+  const skillsOptions = ["Java", "Python", "JavaScript", "HTML", "CSS", "React", "SQL", "PostgreSQL", "AWS", "Azure", "MS Office", "Spring"];
+  const typeOptions = ["Data", "Backend", "Frontend", "Devops", "Cybersecurity", "Cloud", "AI", "Normal"];
+
   const filteredUserCards = users.filter((user) => {
-    if (searchValue) {
-      return user.name.toLowerCase().startsWith(searchValue.toLowerCase());
+    if (searchValue && skillsFilterValue && typesFilterValue) {
+      return user.name.toLowerCase().startsWith(searchValue.toLowerCase()) && user.skills.includes(skillsFilterValue) && user.type.includes(typesFilterValue);
```

Comment:
Storing this in a variable and then iteratively applying the filters will fix the filtering issue. I'll push up a pull request for this fix real quick.